contracts/Insurance.sol

```diff
@@ -142,24 +163,28 @@ contract Insurance is IInsurance, Ownable, SafetyWithdraw {
             return;
         }
 
-        // Enforce a maximum at poolAmount
-        if (amount > collateralAmount) {
-            amount = collateralAmount;
+        if (amount > collateralAmount + bufferAmount) {
+            // Drain both public and buffer insurance pools if needed completely
+            amount = collateralAmount + bufferAmount;
+            collateralAmount = 0;
+            bufferAmount = 0;
+        } else if (amount > bufferAmount) {
+            // Drain buffer insurance pool completely, and then drain part of public insurance pool
+            collateralAmount = collateralAmount + bufferAmount - amount;
+            bufferAmount = 0;
+        } else {
+            // Only need to take part of buffer pool out
+            bufferAmount = bufferAmount - amount;
         }
 
-        // What the balance will be after
-        uint256 difference = collateralAmount - amount;
-        if (difference < 10**18) {
+        if (bufferAmount + collateralAmount < 10**18) {
             // Once we go below one token, social loss is required
             // This calculation caps draining so pool always has at least one token
-            amount = collateralAmount - (10**18);
-            // Use new amount to compute difference again.
-            difference = collateralAmount - amount;
+            bufferAmount = 10**18;
```

Comment:
I think this is wrong. You can't just arbitrarily set bufferAmount to 1. Where is this token coming from?

Reply:
This is making the making sure there is at least one token in the insurance pool, but it should probably be a part of the public pool. I'm gonna speak to Webb about this and see how it should be specc'd

Reply:
yep, as discussed, get rid - no need for this. Only for public account

Reply:
Fixed